Suggestions and Edits from Docs Team (#13)
hasura/supergraph-io: README.md

# The Supergraph Manifesto

## Introduction

> In data API design, a Supergraph is an API development methodology that offers reference specifications, design 
> principles and an operating model. By implementing a Supergraph methodology, multiple teams in an organization can 
> easily collaborate to deliver a modular, interconnected resource of data and logic to create a single, powerful, 
> self-serve endpoint for API consumers.

Reviewer: I feel like we can simplify this to make it easier to grok. Also, we've made a decision to internally use `supergraph` instead of `Supergraph`. Are we giving the concept here the proper-noun treatment?

```suggestion
A Supergraph is a method for designing data APIs. It provides guidelines and best practices for developing APIs. By using the Supergraph approach, different teams in an organization can work together to create a unified, modular system of data and logic. This results in a single, powerful endpoint that API consumers can easily use.
```

## Table of Definitions
| Term               | Definition                                                                                                                     |
|--------------------|--------------------------------------------------------------------------------------------------------------------------------|
| **Supergraph**     | A Supergraph is an API creation methodology that offers reference specifications, design principles and an operating model.    |

> | **supergraph API** | A modular, interconnected resource of data and logic as a single, powerful, self-serve API endpoint.                           |

Reviewer: I see how the casing here relates to my first comment. However, It also looks a bit out-of-place by being the only lowercased item in this table.

> | **Engine**         | Often referred to as a gateway or router in a GraphQL federation stack, managing API requests and responses.                   |

Reviewer: This is the first time we've mentioned GraphQL. It kind of jumps out as a surprise; could we add something in the introduction that addresses how GraphQL fits into this model?

> | **Data Domain**    | A distinct area of functionality or data.                                                                                      |

Reviewer: I think some quick examples will help a user grok this concept.

```suggestion
| **Data Domain**    | A distinct area of functionality or data, such as Customer Management, Inventory Management, or Payment Processing                                                                                      |
```

| **Subgraph**       | A modular component that acts as a self-contained entity within a larger supergraph and often represents a single data domain. |
| **API Consumer**   | An entity that uses APIs to access data and functionality.                                                                     |
| **API Producer**   | An entity that creates and maintains APIs.                                                                                     |

## Before the Supergraph
In conventional API design, data consumers need to manually access, integrate, and compose data from multiple 
endpoints to get the data they need. This is time-consuming, error-prone, and requires deep domain knowledge, often 

> resulting in brittle integrations.

Reviewer: This is great! ❤️

Infrastructure must also be built and maintained to support each of these data-access processes, and individual 
routes or resolvers built and maintained for each data access requirement.

![supergraphioERD.png](assets/supergraphioERD.png)

## Benefits
A Supergraph provides the following key benefits:

1. **Self-Serve API Consumer Composition**: Enables API integration, orchestration, and aggregation in a self-serve 
   manner.
2. **Federated Data Access Layer**: Provides a federated data layer that allows real-time access to data sources with
   cross-domain composability (joins, filtering, etc.).
3. **Incremental Adoption**: Offers a stable API that supports zero-downtime and incremental adoption.

## Supergraph Lifecycle

![ConnectConsumeDiscoverERD.png](assets/ConnectConsumeDiscoverERD.png)

### Connect Domains
As an API producer, you should easily and seamlessly connect your data domains to the platform. The Supergraph strategy 
and architecture addresses common challenges and reluctance from domain owners by ensuring:

> 1. **Subgraph Connector CI/CD**: Keeps the API contract in sync with minimal overhead for the domain owner.

Reviewer: As we haven't introduced the concept of a connector, I think it's extra overhead to mention it here. Above, we're saying that a user should be able to connect seamlessly — this means the connector — once it's deployed — is forgotten about. What they then care about modifying and updating is the subgraph itself and its performance.

```suggestion
1. **Subgraph CI/CD**: Keeps the API contract in sync with minimal overhead for the domain owner.
```

> 2. **Subgraph Connector Performance**: Maintains or improves performance compared to direct access to the underlying 

Reviewer: Same as previous.

```suggestion
2. **Subgraph Performance**: Maintains or improves performance compared to direct access to the underlying 
```
